source/Halibut.TestProxy/ProxyConnection.cs

```diff
@@ -26,7 +25,7 @@ public ProxyConnection(ProxyEndpoint destinationEndpoint, ILogger<ProxyConnectio
             this.logger = logger;
         }
 
-        public async Task Connect(TcpClient source, CancellationToken cancellationToken)
+        public void Connect(TcpClient source, CancellationToken cancellationToken)
         {
```

Comment:
nit: would adding `await Task.CompletedTask;` have made the warning go away?

Reply:
It would have. But I didn't for the reasons outlined in [this comment](https://github.com/OctopusDeploy/Halibut/pull/562#discussion_r1409993158)